plugins/output/flight/flight.go

```diff
@@ -0,0 +1,156 @@
+package flight
+
+import (
+	"context"
+	_ "embed"
+	"log"
+	"net"
+
+	"github.com/apache/arrow/go/v9/arrow"
+	"github.com/apache/arrow/go/v9/arrow/array"
+	"github.com/apache/arrow/go/v9/arrow/flight"
+	"github.com/apache/arrow/go/v9/arrow/ipc"
+	"github.com/apache/arrow/go/v9/arrow/memory"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/outputs"
+)
+
+// DO NOT REMOVE THE NEXT TWO LINES! This is required to embed the sampleConfig data.
+//
+//go:embed sample.conf
+var sampleConfig string
+
+func (*Flight) SampleConfig() string {
+	return sampleConfig
+}
+
+type Flight struct {
+	Location string `toml:"location"`
+	Port     string `toml:"port"`
+	Table    string `toml:"table"`
+
+	timeStamps []arrow.Timestamp
+	values     []float32
+
+	client flight.FlightServiceClient
+	writer flight.Writer
+	schema arrow.Schema
+	ctx    context.Context
+	pool   memory.GoAllocator
+	desc   *flight.FlightDescriptor
+}
+
+// Write should write immediately to the output, and not buffer writes
+// (Telegraf manages the buffer for you). Returning an error will fail this
+// batch of writes and the entire batch will be retried automatically.
+func (s *Flight) Write(metrics []telegraf.Metric) error {
+
+	for _, m := range metrics {
+
+		timeInt := m.Time().UnixMilli()
+
+		s.timeStamps = append(s.timeStamps, arrow.Timestamp(timeInt))
+
+		for _, field := range m.FieldList() {
+			s.values = append(s.values, float32(field.Value.(float64)))
+		}
+
+	}
+
+	b := array.NewRecordBuilder(&s.pool, &s.schema)
+	defer b.Release()
+
+	b.Field(0).(*array.Int32Builder).AppendValues([]int32{1}, nil)
+	b.Field(1).(*array.TimestampBuilder).AppendValues(s.timeStamps, nil)
+	b.Field(2).(*array.Float32Builder).AppendValues(s.values, nil)
+
+	rec := b.NewRecord()
+	defer rec.Release()
+
+	err := s.writer.Write(rec)
+
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	s.timeStamps = nil
+
+	s.values = nil
+
+	s.desc = nil
+
+	return nil
+}
+
+// Init is for setup, and validating config.
+func (s *Flight) Init() error {
+
+	return nil
+}
+
+// Make any connection required here
```

Comment:
Maybe expand this doc comment to document what the method does and add a punctuation at the end.

Reply:
I don't understand this, the doc comment clearly states what the purpose of the method is. I have added punctuation.

Reply:
I would have expected the comment to be something like "Connects to the Apache Arrow Flight server, retrieves the schema for the table to ingest metrics into, and initialize a DoPut stream." However, since the method is not that long and contains comments it is probably fine to leave the short comment as it matches the comment for `Close()`.

Reply:
Fixed.